vignettes/Biostrings2Classes.Rmd

```diff
@@ -0,0 +1,262 @@
+---
+title: "The *Biostrings* 2 classes (work in progress)"
+author: 
+- name: "Hervé Pagès"
+- name: "Paul Villafuerte"
+  affiliation: "Vignette translation from Sweave to Rmarkdown / HTML"
+date: "`r format(Sys.time(), '%B %d, %Y')`"
+vignette: >
+  %\VignetteIndexEntry{A short presentation of the basic classes defined in Biostrings 2}
+  %\VignetteKeywords{DNA, RNA, Sequence, Biostrings, Sequence alignment}
+  %\VignettePackage{Biostrings}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+output:
+  BiocStyle::html_document:
+    number_sections: true
+    toc: yes
+    toc_depth: 4
+editor_options: 
+  markdown:   
+    wrap: 72 
+---
+  
+# Introduction
+
+This document briefly presents the new set of classes implemented in the
+`r Biocpkg('Biostrings')` 2 package. Like the `r Biocpkg('Biostrings')` 
+1 classes (found in `r Biocpkg('Biostrings')` v 1.4.x), they were designed to make manipulation of big
+strings (like DNA or RNA sequences) easy and fast. This is achieved by
+keeping the 3 following ideas from the `r Biocpkg('Biostrings')` 1 package: (1) use R
+external pointers to store the string data, (2) use bit patterns to
+encode the string data, (3) provide the user with a convenient class of
+objects where each instance can store a set of views *on the same* big
+string (these views being typicallythe matches returned by a search
+algorithm).
+
+However, there is a flaw in the `r Biocpkg('Biostring')` class design that prevents
+the search algorithms to return correct information about the matches
+(i.e. the views) that they found. The new classes address this issue by
+replacing the `r Biocpkg('Biostrings')` class (implemented in `r Biocpkg('Biostrings')` 1) by 2 new
```

Comment:
I think lines 37 and 40 (the first reference only) reference an old `BioString` class, so replace

    `r Biocpkg('Biostrings')`
    
with

    BioString